src/pudl/extract/excel.py

```diff
@@ -67,24 +67,24 @@ def get_dataset_name(self):
         """Returns the name of the dataset described by this metadata."""
         return self._dataset_name
 
-    def get_sheet_name(self, year, page):
-        """Returns name of the excel sheet that contains the data for given year and page."""
-        return self._sheet_name.at[page, str(year)]
+    def get_sheet_name(self, page, **partition):
+        """Returns name of the excel sheet that contains the data for given partition and page."""
+        return self._sheet_name.at[page, str(list(partition.values())[0])]
```

Comment:
this feels like a hack.... is there a better way to unpack the kwarg here? at this stage it should always be setup as partition_name=partition (`year=2019` or `year_month="2020-08"`)

Reply:
This indeed feels like a hack. The implicit assumption here is that `**partition` contains exactly one `key=value` pair and we use the stringified value when pulling the sheet name from pandas data frame.

I would make this assumption explicit in the form of `AssertonError`:
```
if len(partition) != 1:
  raise AssertionError(f"Expecting exactly one partition attribute (found: {partition})")
return self._sheet_name.at[page, str(partition.values()[0])]
```

Reply:
I now see that this value extraction happens multiple time so I would suggest creating helper function `_get_partition_key(**partitions)` that will do that.

Such approach could easily be modified if we ever encounter multi-dimensional partitions (e.g. year, month) where you could construct keys in the form of `"month=1,year=2019"` and use these for indexing pandas dataframes.

Reply:
this is a great idea!